app/src/org/commcare/dalvik/provider/FixtureDataContentProvider.java

```diff
@@ -0,0 +1,179 @@
+package org.commcare.dalvik.provider;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.commcare.android.util.CommCareInstanceInitializer;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.javarosa.core.model.instance.DataInstance;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.storage.IStorageIterator;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.model.xform.DataModelSerializer;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.MatrixCursor;
+import android.net.Uri;
+
+/**
+ * The case data content provider defines the interface for external applications
+ * to gain read only access to the current user's sandbox. External applications require
+ * explicit permissions to access the content provider, and only data in the currently
+ * logged in user's sandbox is provided.
+ * 
+ * The CaseDataAPI class is responsible for defining the data models and structures that
+ * can be queried with the content provider.
+ * 
+ * Note that this content provider requires a permission to be granted for read access.
+ * 
+ * No write access is currently supported.
+ * 
+ * @author ctsims
+ *
+ */
+public class FixtureDataContentProvider extends ContentProvider {
+
+    /* (non-Javadoc)
+     * @see android.content.ContentProvider#onCreate()
+     */
+    @Override
+    public boolean onCreate() {
+
+        return true;
+    }
+
+    @Override
+    public Cursor query(Uri uri, String[] projection, String selection,
+            String[] selectionArgs, String sortOrder) {
+
+        IStorageUtilityIndexed<FormInstance> userFixtureStorage;
+
+        try{
+            userFixtureStorage = CommCareApplication._().getUserStorage("fixture", FormInstance.class);
+        } catch(SessionUnavailableException sue){
+            return null;
+        }
+
+        //Standard dispatcher following Android best practices
+        int match = FixtureDataAPI.UriMatch(uri);
```

Comment:
is there a reason why we're using a couple different paths here? I'd imagine that we could just use one URI, and then we could just use the Projection to see which columns they want included.